client/src/main/java/io/pravega/client/segment/impl/SegmentMetadataClientImpl.java

```diff
@@ -105,26 +115,37 @@ RawClient getConnection() {
             throw new NoSuchSegmentException(reply.toString());
         } else if (reply instanceof WrongHost) {
             throw new ConnectionFailedException(reply.toString());
+        } else if (reply instanceof WireCommands.AuthTokenCheckFailed) {
+            WireCommands.AuthTokenCheckFailed authTokenCheckReply = (WireCommands.AuthTokenCheckFailed) reply;
+            if (authTokenCheckReply.isTokenExpired()) {
+                log.info("Delegation token expired");
+                throw new TokenExpiredException(authTokenCheckReply.toString());
+                // closeConnection(reply);
+            } else {
+                log.info("Delegation token invalid");
+                throw new TokenException(authTokenCheckReply.toString());
+            }
         } else {
             throw new ConnectionFailedException("Unexpected reply of " + reply + " when expecting a "
                     + klass.getName());
         }
     }
 
-    private CompletableFuture<StreamSegmentInfo> getStreamSegmentInfo(String delegationToken) {
+    private CompletableFuture<StreamSegmentInfo> getStreamSegmentInfo(DelegationTokenProxy delegationToken) {
         log.debug("Getting segment info for segment: {}", segmentId);
         RawClient connection = getConnection();
         long requestId = connection.getFlow().getNextSequenceNumber();
-        return connection.sendRequest(requestId, new GetStreamSegmentInfo(requestId, segmentId.getScopedName(), delegationToken))
+        return connection.sendRequest(requestId, new GetStreamSegmentInfo(
+                requestId, segmentId.getScopedName(), delegationToken.retrieveToken()))
                          .thenApply(r -> transformReply(r, StreamSegmentInfo.class));
     }
     
-    private CompletableFuture<WireCommands.SegmentAttribute> getPropertyAsync(UUID attributeId, String delegationToken) {
+    private CompletableFuture<WireCommands.SegmentAttribute> getPropertyAsync(UUID attributeId, DelegationTokenProxy delegationToken) {
```

Comment:
Same here

Reply:
This was preexisting code, where I had replaced the String argument with a `DelegationTokenProvider` argument. In any case, I have done the change as suggested.